### DISCUSSION TITLE
How can I set the working directory after splitting a screen?

### DISCUSSION THREAD

**Discussion by toomuat** [wezterm/wezterm#2849] [Q&A]
When I split the pane using SplitVertical or SplitHorizontal, a new window is created in the directory set in default_cwd.
How can I make the working directory of the split destination the same as that of the split source? It is a little troublesome to move to the previous directory after split.

The current settings for splitting screen are as follows
```
        { key = "-", mods = "LEADER",       action=wezterm.action{SplitVertical={domain="CurrentPaneDomain"}}},
        { key = "\\",mods = "LEADER",       action=wezterm.action{SplitHorizontal={domain="CurrentPaneDomain"}}}
```

**Comment by wez** [comment]
https://wezfurlong.org/wezterm/config/lua/config/default_cwd.html explains how the current directory is determined when spawning a new pane.

**Comment by ashwinvis** [comment]
TLDR:
- wezterm recognizes the cwd using OSC 7 (https://wezfurlong.org/wezterm/shell-integration.html)
- For bash and zsh, you can [download a script ](https://github.com/wez/wezterm/tree/main/assets/shell-integration) and add `source path/to/wezterm.sh` in your bashrc / zshrc to enable this
